packages/app-desktop/gui/NoteEditor/NoteEditor.tsx

```diff
@@ -65,9 +65,21 @@ function NoteEditor(props: NoteEditorProps) {
 		setTitleHasBeenManuallyChanged(false);
 	}, []);
 
+	// When a notebook is changed without any selected note,
+	// no note is selected for a moment, and then a new note gets selected.
+	// In this short transient period, the last displayed note id should temporarily
+	// be used to prevent NoteEditor's body from being unmounted.
+	const lastDisplayedNoteId = useRef<string>(null);
+	const whenNoteIdIsTransientlyAbsent = !props.noteId && props.notes.length > 0;
+	const effectiveNoteId = whenNoteIdIsTransientlyAbsent ? lastDisplayedNoteId.current : props.noteId;
+
+	useEffect(() => {
+		if (props.noteId) lastDisplayedNoteId.current = props.noteId;
+	}, [props.noteId]);
```

Comment:
@ken1kob, would you mind wrapping all this logic in a hook like `const effectiveNoteId = useEffectiveNoteId(....)`? Other than that, I think we can merge. I'm not a huge fan of this trick but, as you identified there's no other ways to do it, so let's go with it. At least if we can wrap the hack in a hook it will be a bit easier to see what's going on.

Reply:
Finished.